pystac/catalog.py

```diff
@@ -705,15 +705,24 @@ def generate_subcatalogs(
 
         return result
 
-    def save(self, catalog_type: Optional[CatalogType] = None) -> None:
+    def save(
+        self,
+        catalog_type: Optional[CatalogType] = None,
+        dest_href: Optional[str] = None,
+    ) -> None:
         """Save this catalog and all it's children/item to files determined by the object's
-        self link HREF.
+        self link HREF or a specified path.
 
         Args:
             catalog_type : The catalog type that dictates the structure of
                 the catalog to save. Use a member of :class:`~pystac.CatalogType`.
                 If not supplied, the catalog_type of this catalog will be used.
                 If that attribute is not set, an exception will be raised.
+            dest_href : The location where the catalog is to be saved.
+                If not supplied, the catalog's self link HREF is used to determine
+                the location of the catalog file and children's files.
+                This is olnly used if the catalog type is
+                ``CatalogType.ABSOLUTE_PUBLISHED``.
```

Comment:
It might be good to add a check at the beginning of this method that raises a `STACError` if `dest_href` is not `None` and `root.catalog_type != CatalogType.ABSOLUTE_PUBLISHED`. I could see users being confused if they pass in a `dest_href` that just gets ignored silently.

Reply:
I restricted it to `CatalogType.ABSOLUTE_PUBLISHED`, because the original issue in the stactools repo mentioned the functionality for that case, but it's true that here it makes sense for other types as well (If anyone has a different opinion here or I am missing something, please comment here). I have removed that check and now the `dest_href` parameter is used regardless of the catalog type